roles/kubernetes/common/tasks/main.yaml

```diff
@@ -37,16 +37,42 @@
 
 - name: Enable cgroup v2 # https://rootlesscontaine.rs/getting-started/common/cgroup2/#enabling-cgroup-v2
   block:
-    - name: Adding GRUB_CMDLINE_LINUX to /etc/default/grub file
+    - name: Retrieve contents of /etc/default/grub
+      ansible.builtin.shell: cat /etc/default/grub
+      register: grub_content
+      changed_when: False
+
+    - name: Adding GRUB_CMDLINE_LINUX to /etc/default/grub if absent
       ansible.builtin.lineinfile:
         path: /etc/default/grub
-        regexp: '^GRUB_CMDLINE_LINUX='
-        line: GRUB_CMDLINE_LINUX="systemd.unified_cgroup_hierarchy=1"
+        line: 'GRUB_CMDLINE_LINUX="{{ option }}={{ value }}"'
         owner: root
         group: root
         mode: '0644'
+      when: '"GRUB_CMDLINE_LINUX=" not in grub_content.stdout'
+
+    - name: Adding {{ option }}=={{ value }} to existing GRUB_CMDLINE_LINUX in /etc/default/grub
+      ansible.builtin.lineinfile:
+        path: /etc/default/grub
+        backrefs: yes
+        regexp: '^(GRUB_CMDLINE_LINUX=(?!.*{{ option | regex_escape }})"[^"]*)(")'
+        line: '\1 {{ option }}={{ value }}\2'
```

Comment:
Just making sure the space between `\1` and the `{{` brackets was intentional. I am guessing this is like an append to the end of the existing command line.

Reply:
I would use a regex or chomp like function to eat the leading trailing spaces if possible.

Reply:
I couldn't figure out how to make backrefs work with ansible filters, so the code has been refactored. It should no longer include the leading space when updating an empty string, and will also now work when GRUB_CMDLINE_LINUX is in the file but commented out.